src/serializer/ResidualFunctions.js

```diff
@@ -223,7 +225,11 @@ export class ResidualFunctions {
     for (let [funcBody, instances] of functionEntries) {
       let functionInfo = this.residualFunctionInfos.get(funcBody);
       invariant(functionInfo);
-      this._referentialize(functionInfo.unbound, instances);
+      // TODO: figure out how referentialization works with additional functions
```

Comment:
Referentialization kicks in on modified captured bindings.
To make things safe, check if any rewritten additional functions have any modified serialized bindings, and if so, log an error.